### ISSUE TITLE
CSV waits for wrong CRD version to be available

### ISSUE THREAD

**Issue by rhuss** [operator-framework/operator-lifecycle-manager#507]
I did my first steps with the OLM and experimented a bit by writing my own CSV. In my first attempt I got the CRD version wrong, so I used '1beta1' instead of '1alpha1'. So I removed the CSV again and re-applied it with the changed version. 

However I first wast stuck in `RequirementsNotMet` condition with the OLM still waiting on previously specified CSV, but now I'm this state for the CSV:

```yaml
// ...
spec:
  customresourcedefinitions:
    owned:
    - description: Camel in the cloud
      displayName: Syndesis
      kind: Syndesis
      name: syndesises.syndesis.io
      specDescriptors:
      - description: Specify how many integrations are allowed for this installation
        displayName: Integration Limit
        path: integration.limit
        x-descriptors:
        - urn:alm:descriptor:com.tectonic.ui:label
      version: v1alpha1
// ....
status:
 /// ...
 message: install strategy completed with no errors
  phase: Succeeded
  reason: InstallSucceeded
  requirementStatus:
  - group: apiextensions.k8s.io
    kind: CustomResourceDefinition
    name: syndesises.syndesis.io
    status: Present
    uuid: a76cbe04-cc6a-11e8-acbc-0ed2eff6cb2c
    version: v1beta1
```

which actually says that version v1beta1 is present (but configured is v1alpha1 in the spec)

Also, there is only a 1alpha1 CRD registered in the cluster (`oc get crd syndesises.syndesis.io -o json | jq .spec`)

```json
{
  "group": "syndesis.io",
  "names": {
    "kind": "Syndesis",
    "listKind": "SyndesisList",
    "plural": "syndesises",
    "singular": "syndesis"
  },
  "scope": "Namespaced",
  "version": "v1alpha1"
}
```

Could it be that the `requirementStatus` uses some stale cached version ?

**Comment by njhale** [comment]
The version v1beta1 listed in the requirementStatus block is the version of apiextensions.k8s.io CustomResourceDefinition rather than the version of your specific CRD. Currently OLM does not distinguish between specific CRD versions (we're planning on implementing that in the future).

**Comment by rhuss** [comment]
@njhale ah, thanks for the clarification. Now everything makes sense ;-)

However, I had the feeling when I deployed a `1alpha` version of my CRD but specified a `1beta1` in the CSV, then the OLM didn't pick up my CRD as being a valid requirement fulfilment. I have to recheck though.

**Comment by njhale** [comment]
@rhuss Here's a [PR](https://github.com/operator-framework/operator-lifecycle-manager/pull/514) that attempts to address the CRD version requirement check.

**Comment by ecordell** [comment]
Fixed in #514
